linode/helper/objects.go

```diff
@@ -99,6 +99,15 @@ func IsObjNotFoundErr(err error) bool {
 	return errors.As(err, &apiErr) && (apiErr.ErrorCode() == "NotFound" || apiErr.ErrorCode() == "Forbidden")
 }
 
+// isBucketNotFoundError checks if the error is due to the bucket not being found.
+func IsBucketNotFoundError(err error) bool {
+	tflog.Debug(
+		context.Background(),
+		fmt.Sprintf("received an error: %s, checking whether it's a bucket not found error", err),
+	)
+	return strings.Contains(err.Error(), "[400] [bucket_access[0].bucket_name] Bucket not found")
```

Comment:
Is it possible the index would not be `0`? Like `[400] [bucket_access[998].bucket_name] Bucket not found`

Reply:
I didn't consider that. I suppose that it is possible so I can update it to just check for "Bucket not found" instead.

Reply:
Fixed!